sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/filesystem/DefaultModuleFileSystem.java

```diff
@@ -46,22 +46,26 @@
   private List<File> testDirsOrFiles = Lists.newArrayList();
   private ComponentIndexer componentIndexer;
   private boolean initialized;
+  private Charset charset = null;
 
   public DefaultModuleFileSystem(ModuleInputFileCache moduleInputFileCache, Project project,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)